### PR TITLE
Feature/2010  add build version parameter as token for frontend served file

### PR DESCRIPTION
**Pull request by inkhey**
related to #2010 
require  #2619
## Checkpoints

The goal of this section is to help code reviewer to do required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

- [x] Code is clear enough
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

- [x] Tested by quality team

### CODE REVIEW

File: backend/tracim_backend/lib/utils/utils.py [hunk 1]
```diff
@@ -416,3 +417,25 @@ def find_all_submodule_path(module: types.ModuleType) -> typing.List[str]:
...
+    except (ImportError, InvalidGitRepositoryError):
+        return "unknown"
+
+    tags = [tag for tag in repo.tags if tag.commit == repo.head.commit]
+    if tags:
+        # INFO - G.M - 2020-01-13 - return first associated tag to head commit
+        return tags[0]
+    else:
+        # INFO - G.M - 2020-01-13 - return the 10 first letter of current commit hash
+        return repo.head.object.hexsha[:10]
```
Comment (buxx) [inline backend/tracim_backend/lib/utils/utils.py]: Another way to do this:

```python
try:
    return next((tag for tag in repo.tags if tag.commit == repo.head.commit))
except StopIteration:
    return repo.head.object.hexsha[:10]
```

File: backend/tracim_backend/views/core_api/schemas.py [hunk 2]
```diff
@@ -966,7 +966,10 @@ class TimezoneSchema(marshmallow.Schema):
 
 class AboutSchema(marshmallow.Schema):
     name = StrippedString(example="Tracim", description="Software name")
-    version = StrippedString(example="2.0", allow_none=True, description="Version of Tracim")
+    version = StrippedString(example="2.6", allow_none=True, description="Version of Tracim")
+    build_version = StrippedString(
+        example="release_02.06.00", allow_none=True, description="Build Version of Tracim"
+    )
```
Comment (buxx) [inline backend/tracim_backend/views/core_api/schemas.py]: Why `version` and `build_version` can be `None` ?

File: backend/tracim_backend/views/core_api/schemas.py [hunk 3]
```diff
@@ -966,7 +966,10 @@ class TimezoneSchema(marshmallow.Schema):
...
     datetime = marshmallow.fields.DateTime(format=DATETIME_FORMAT)
     website = marshmallow.fields.URL(allow_none=True)
```
Comment (buxx) [inline backend/tracim_backend/views/core_api/schemas.py]: Same question for `website`

File: frontend/dist/index.mak [hunk 4]
```diff
@@ -77,33 +77,33 @@
...
+    <script type='text/javascript' src='/app/admin_workspace_user.app.js?v=${build_version}'></script>
+    <script type='text/javascript' src='/app/workspace_advanced.app.js?v=${build_version}'></script>
+    <script type='text/javascript' src='/app/gallery.app.js?v=${build_version}'></script>
 
-    <script type='text/javascript' src='/assets/bootstrap/jquery-3.2.1.js'></script>
-    <script type='text/javascript' src='/assets/bootstrap/popper-1.12.3.js'></script>
-    <script type='text/javascript' src='/assets/bootstrap/bootstrap-4.0.0-beta.2.js'></script>
+    <script type='text/javascript' src='/assets/bootstrap/jquery-3.2.1.js?v=${build_version}'></script>
+    <script type='text/javascript' src='/assets/bootstrap/popper-1.12.3.js?v=${build_version}'></script>
+    <script type='text/javascript' src='/assets/bootstrap/bootstrap-4.0.0-beta.2.js?v=${build_version}'></script>
```
Comment (buxx) [inline frontend/dist/index.mak]: @Skylsmoi Does it be required on these javascript files (they have version in name)

**Comment by inkhey** [comment]
> ```python
> BUILD_VERSION
> ```

Problem with this solution, is that build version has to be updated at each commit  or we get incorrect information. I do think it's a heavy mechanism.

This dual version mecanism : version and build_version allow us to have exact build whenever is it possible and to set him through config_file/env_var if we need it (for example in docker context without git).

File: backend/tracim_backend/views/core_api/schemas.py (hunk 3 above)
Comment (inkhey) [reply backend/tracim_backend/views/core_api/schemas.py]: fixed.

File: backend/tracim_backend/views/core_api/schemas.py (hunk 2 above)
Comment (inkhey) [reply backend/tracim_backend/views/core_api/schemas.py]: fixed.

File: backend/tracim_backend/lib/utils/utils.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/lib/utils/utils.py]: done.

File: frontend/dist/index.mak (hunk 4 above)
Comment (Skylsmoi) [reply frontend/dist/index.mak]: you're right, it isn't
If we change the library version, it'll be reflected in the filename

Comment (inkhey) [reply frontend/dist/index.mak]: fixed.

**Comment by inkhey** [comment]
:warning: this require docker support before adding to test.

**Comment by PhilippeAccorsi** [comment]
Tested on branch and with new docker images.

**Pull request merged**